### PR TITLE
Small usability improvement for `slic::SimpleLogger`

### PR DESCRIPTION
**Pull request by kennyweiss**
# Summary

- This PR adds a small usability tweak to `slic::SimpleLogger` and fixes up its usage throughout our tests and examples
- It allow one to set the initial logging level in the constructor for SimpleLogger
- I had this improvement in a feature branch that's not yet ready to merge and wanted to use it in another branch

### CODE REVIEW

File: src/axom/slic/core/SimpleLogger.hpp [hunk 1]
```diff
@@ -60,10 +61,10 @@ class SimpleLogger
   /*!
    * \brief Constructor initializes slic logging environment.
    */
-  SimpleLogger()
+  explicit SimpleLogger(message::Level level = message::Debug)
```
Comment (kennyweiss) [inline src/axom/slic/core/SimpleLogger.hpp:64]: This line contains the main change in this PR -- it adds an initial logging level for the `SimpleLogger`.

File: src/axom/spin/tests/spin_rectangular_lattice.cpp [hunk 2]
```diff
@@ -679,12 +678,7 @@ int main(int argc, char* argv[])
   int result = 0;
 
   ::testing::InitGoogleTest(&argc, argv);
-
-  using axom::slic::SimpleLogger;
-  SimpleLogger logger;  // create & initialize test logger,
-  axom::slic::setLoggingMsgLevel(axom::slic::message::Info);
-
-  // finalized when exiting main scope
+  axom::slic::SimpleLogger logger(axom::slic::message::Info);
```
Comment (kennyweiss) [inline src/axom/spin/tests/spin_rectangular_lattice.cpp:681]: This is a prototypical usage that motivated the PR

File: src/axom/slam/tests/slam_relation_DynamicVariable.cpp [hunk 3]
```diff
@@ -213,10 +213,7 @@ TEST(slam_relation_dynamic_variable, iterate_relation)
 int main(int argc, char* argv[])
 {
   ::testing::InitGoogleTest(&argc, argv);
-
-  // create & initialize test logger. finalized when exiting main scope
-  axom::slic::SimpleLogger logger;
-  axom::slic::setLoggingMsgLevel(axom::slic::message::Info);
+  axom::slic::SimpleLogger logger(axom::slic::message::Info);
```
Comment (white238) [inline src/axom/slam/tests/slam_relation_DynamicVariable.cpp:216]: Why would a test not want debug messages?

File: src/axom/sidre/examples/sidre_generateindex.cpp [hunk 4]
```diff
@@ -40,29 +41,31 @@
 #include <cstring>
 
 // "using" directives to simplify code
-using namespace axom;
-using namespace sidre;
+namespace sidre = axom::sidre;
```
Comment (white238) [inline src/axom/sidre/examples/sidre_generateindex.cpp:44]: 👍

File: src/axom/mint/examples/mint_unstructured_mixed_topology_mesh.cpp [hunk 5]
```diff
@@ -29,7 +28,7 @@ inline bool appendQuad(axom::IndexType i, axom::IndexType j)
 //------------------------------------------------------------------------------
 int main(int AXOM_UNUSED_PARAM(argc), char** AXOM_UNUSED_PARAM(argv))
 {
-  SimpleLogger logger;  // create & initialize test logger,
+  axom::slic::SimpleLogger logger;  // create & initialize test logger,
```
Comment (white238) [inline src/axom/mint/examples/mint_unstructured_mixed_topology_mesh.cpp:31]:
```suggestion
  axom::slic::SimpleLogger logger;  // create & initialize test logger
```

**Review by white238 — APPROVED** [review]

**Comment by white238** [comment]
Thanks for keeping the PR's concise!

**Review by rhornung67 — APPROVED** [review]
Thanks @kennyweiss 👍

File: src/axom/mint/examples/mint_unstructured_single_topology_mesh.cpp [hunk 6]
```diff
@@ -16,12 +16,11 @@
 #include "axom/slic.hpp"
 
 using namespace axom;
-using axom::slic::SimpleLogger;
 
 //------------------------------------------------------------------------------
 int main(int AXOM_UNUSED_PARAM(argc), char** AXOM_UNUSED_PARAM(argv))
 {
-  SimpleLogger logger;  // create & initialize test logger,
+  axom::slic::SimpleLogger logger;  // create & initialize test logger,
```
Comment (agcapps) [inline src/axom/mint/examples/mint_unstructured_single_topology_mesh.cpp:23]:
```suggestion
  axom::slic::SimpleLogger logger;  // create & initialize test logger
```

File: src/axom/primal/tests/primal_clip.cpp [hunk 7]
```diff
@@ -655,14 +656,10 @@ TEST(primal_clip, oct_tet_clip_special_case_2)
...
 //------------------------------------------------------------------------------
-#include "axom/slic/core/SimpleLogger.hpp"
-using axom::slic::SimpleLogger;
-
 int main(int argc, char* argv[])
 {
   ::testing::InitGoogleTest(&argc, argv);
-
-  SimpleLogger logger;  // create & initialize test logger,
+  axom::slic::SimpleLogger logger;  // create & initialize test logger,
```
Comment (agcapps) [inline src/axom/primal/tests/primal_clip.cpp:662]:
```suggestion
  axom::slic::SimpleLogger logger;
```

File: src/axom/quest/Delaunay.hpp [hunk 8]
```diff
@@ -74,7 +74,7 @@ class Delaunay
 public:
   /**
    * \brief Default constructor
-   * \note User must to call initializeBoundary(BoundingBox) before adding points.
+   * \note User must call initializeBoundary(BoundingBox) before adding points.
```
Comment (agcapps) [inline src/axom/quest/Delaunay.hpp:77]: Thank you for catching the typo.

File: src/axom/sidre/tests/spio/spio_main.cpp [hunk 9]
```diff
@@ -23,16 +24,12 @@ const std::string ROOT_EXT = ".root";
...
-using axom::slic::SimpleLogger;
-
 int main(int argc, char* argv[])
 {
   int result = 0;
 
   ::testing::InitGoogleTest(&argc, argv);
-
-  SimpleLogger logger;  // create & initialize test logger,
+  axom::slic::SimpleLogger logger;  // create & initialize test logger,
```
Comment (agcapps) [inline src/axom/sidre/tests/spio/spio_main.cpp:32]:
```suggestion
  axom::slic::SimpleLogger logger;
```

**Review by agcapps — APPROVED** [review]
Thanks, Kenny.

File: src/axom/slam/tests/slam_relation_DynamicVariable.cpp (hunk 3 above)
Comment (kennyweiss) [reply src/axom/slam/tests/slam_relation_DynamicVariable.cpp:216]: Not sure -- maybe it was too verbose and hiding what we really wanted to see? I didn't look into it in this PR, I just followed what was already there.

**Pull request merged**